Until bpftrace is updated to use newer BTF functions, we need to use the deprecated form.

See the comment here: https://github.com/libbpf/libbpf/blob/fd181bc349bd6d479cc0a9fdf122885ad352eaf8/src/btf.h#L294-L297

--- a/src/btf.cpp
+++ b/src/btf.cpp
@@ -205,7 +205,7 @@ static struct btf_dump *dump_new(const struct btf *btf,
     .ctx = ctx,
   };
 
-  return btf_dump__new(btf, nullptr, &opts, dump_printf);
+  return btf_dump__new_deprecated(btf, nullptr, &opts, dump_printf);
 #endif
 }
 
